### PR TITLE
Added FileList.ro torrent provider

### PR DESCRIPTION
**Pull request by madpipeline**
I have not tested this, as I don't know how.
- [x] PR is based on the DEVELOP branch
- [x] Don't send big changes all at once. Split up big PRs into multiple smaller PRs that are easier to manage and review
- [x] Read [contribution guide](https://github.com/SickRage/SickRage/blob/master/.github/CONTRIBUTING.md)

### CODE REVIEW

**Comment by SickRage2** [comment]
Thanks for the pull request! Before a real human comes by, please make sure your PR has all of the below criteria checked
- [x] Give a description on what the PR is for.
- [x] Make sure your PR is based on the DEVELOP branch
- [x] Don't send big changes all at once. Split up big PRs into multiple smaller PRs that are easier to manage and review

Please make sure you also read [contribution guide](https://github.com/SickRage/SickRage/blob/master/contributing.md) and followed all of the steps.

Thanks!

**Comment by madpipeline** [comment]
I've tested it and it successfully finds and downloads the desired torrents.

**Comment by madpipeline** [comment]
IMO it can be merged now.

**Comment by miigotu** [comment]
Leave it open for now. I have a few small changes, so later I will pull it, rebase, and fix up the quantifiedCode warnings and a few other small fixes.

Good work

**Comment by madpipeline** [comment]
What's the state of this PR?

**Comment by miigotu** [comment]
Just need a little time to open up for me lol. I have some local changes for it

**Comment by madpipeline** [comment]
@miigotu Can this still me merged? I can look into the merge conflicts, but only if I know I'm not doing this for nothing.

File: sickbeard/providers/filelist.py [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+                data = self.get_url(search_url, params=search_params, returns="text")
+                if not data:
+                    logger.log("No data returned from provider", logger.DEBUG)
+                    continue
+
+                with BS4Parser(data, "html5lib") as html:
+                    torrent_rows = html.find_all("div", class_="torrentrow")
+
+                    # Continue only if at least one Release is found
+                    if len(torrent_rows) < 1:
```
Comment (miigotu) [inline sickbeard/providers/filelist.py]: `if not torrent_rows:`

File: sickbeard/providers/filelist.py [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+            result = ""
+            if td.a and td.a.img:
+                result = td.a.img.get("title", td.a.get_text(strip=True))
+            if not result:
+                result = td.get_text(strip=True)
+            return result
+
+        for mode in search_strings:
+            items = []
+            logger.log("Search Mode: {}".format(mode), logger.DEBUG)
```
Comment (miigotu) [inline sickbeard/providers/filelist.py]: `logger.log("Search Mode: {0}".format(mode), logger.DEBUG)`

File: sickbeard/providers/filelist.py [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+                                continue
+
+                            seeders = try_int(cells[labels.index("Seeders")].find("span").get_text(strip=True))
+                            leechers = try_int(cells[labels.index("Leechers")].find("span").get_text(strip=True))
+
+                            # Filter unseeded torrent
+                            if seeders < self.minseed or leechers < self.minleech:
+                                if mode != "RSS":
+                                    logger.log("Discarding torrent because it doesn't meet the"
+                                               " minimum seeders or leechers: {} (S:{} L:{})".format
```
Comment (miigotu) [inline sickbeard/providers/filelist.py]: `" minimum seeders or leechers: {0} (S:{1} L:{2})".format`

File: sickbeard/providers/filelist.py [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+                                               (title, seeders, leechers), logger.DEBUG)
+                                continue
+
+                            torrent_size = cells[labels.index("Size")].find("span").get_text(strip=True)
+                            size = convert_size(torrent_size, units=units, sep='') or -1
+
+                            item = {'title': title, 'link': download_url, 'size': size, 'seeders': seeders, 'leechers': leechers, 'hash': None}
+                            if mode != "RSS":
+                                logger.log("Found result: {} with {} seeders and {} leechers".format
```
Comment (miigotu) [inline sickbeard/providers/filelist.py]: `logger.log("Found result: {0} with {1} seeders and {2} leechers".format`

**Review by miigotu — CHANGES_REQUESTED** [review]
Conflict is in **init**. Some trivial changes I mentioned below but overall it looks great.
I only mentioned the change suggested by https://www.quantifiedcode.com/app/project/gh:SickRage:SickRage/diff/422fe8ae30eed0bc4c7feedefd5f666011d69440/527225365b9843e38cea2fc0d7138880192af257/issues

File: sickbeard/providers/filelist.py (hunk 2 above)
Comment (madpipeline) [reply sickbeard/providers/filelist.py]: Added.

File: sickbeard/providers/filelist.py (hunk 1 above)
Comment (madpipeline) [reply sickbeard/providers/filelist.py]: Added.

File: sickbeard/providers/filelist.py (hunk 3 above)
Comment (madpipeline) [reply sickbeard/providers/filelist.py]: Added.

File: sickbeard/providers/filelist.py (hunk 4 above)
Comment (madpipeline) [reply sickbeard/providers/filelist.py]: Added.

**Comment by madpipeline** [comment]
I see code coverage has decreased. Can I make some kind of test for this patch?
How would I make that?

**Comment by miigotu** [comment]
Dont worry about the coverage, coveralls doesnt properly detect our provider tests for 2 reasons: 1, the proivders are lazy loaded and 2. One test suite tests all providers.

**Review by miigotu — APPROVED** [review]

**Pull request merged**